Avoid repeated factory creation in media_attachments_vacuum_spec

mjankowski opened the pull request:
Reduces factory creation from 54 to 12, spec run time by ~60%, preserves coverage.

ClearlyClaire commented:
I think I'd rather have all those attachments always be created, but combine the assertions so that there is only one run of the code, but I don't feel strongly either way.

mjankowski committed:
Avoid repeated factory creation in media_attachments_vacuum_spec

mjankowski commented:
Updated to combine into one example and have the before process the vacuum against all records.

Added some comments to preserve a little of the explanation previously in the example descriptions.
